pypesto/hierarchical/petab.py

```diff
@@ -504,3 +521,139 @@ def _get_symbolic_formula_from_measurement(
+                    corresponding_obs
+                    & observables_by_data_type[SEMIQUANTITATIVE]
+                ):
+                    continue
+                relative_observables.update(corresponding_obs)
+
+    observables_by_data_type[RELATIVE] = relative_observables
+
+    # Check whether there is any overlap across data types
+    for data_type, observables in observables_by_data_type.items():
+        for (
+            other_data_type,
+            other_observables,
+        ) in observables_by_data_type.items():
+            if data_type == other_data_type:
+                continue
+            if observables & other_observables:
+                raise ValueError(
+                    "The same observable is associated with multiple data "
+                    f"types. Observable(s): `{observables & other_observables}`. "
+                    f"Data types: `{data_type}`, `{other_data_type}`."
+                )
+
+    # Validate ordinal measurement specification
+    if ORDINAL in observables_by_data_type:
+        meas_df_w_ordinals = meas_df[meas_df[MEASUREMENT_TYPE] == ORDINAL]
+        if MEASUREMENT_CATEGORY not in meas_df_w_ordinals.columns:
+            raise ValueError(
+                "Measurement category must be specified for ordinal "
+                "measurements."
+            )
+        for _, row in meas_df_w_ordinals.iterrows():
+            try:
+                int(row[MEASUREMENT_CATEGORY])
+            except ValueError as e:
+                raise ValueError(
+                    "Measurement category for ordinal measurement must be "
+                    "an integer."
+                ) from e
+
+    # Validate censored measurement specification
+    if set(CENSORING_TYPES) & set(observables_by_data_type.keys()):
+        meas_df_w_censored = meas_df[
+            meas_df[MEASUREMENT_TYPE].isin(CENSORING_TYPES)
+        ]
+        if CENSORING_BOUNDS not in meas_df_w_censored.columns:
+            raise ValueError(
+                "Censoring bounds must be specified for censored "
+                "measurements."
+            )
+        for _, row in meas_df_w_censored.iterrows():
+            try:
+                if (
+                    row[MEASUREMENT_TYPE] == LEFT_CENSORED
+                    or row[MEASUREMENT_TYPE] == RIGHT_CENSORED
+                ):
+                    float(row[MEASUREMENT_CATEGORY])
+                elif row[MEASUREMENT_TYPE] == INTERVAL_CENSORED:
+                    bounds = row[MEASUREMENT_CATEGORY].split(
+                        PARAMETER_SEPARATOR
+                    )
+                    if len(bounds) != 2:
+                        raise ValueError(
+                            f"Censoring bounds for a {INTERVAL_CENSORED} measurement"
+                            f" must be two floats separated by the separator {PARAMETER_SEPARATOR}."
+                        )
+                    float(bounds[0])
+                    float(bounds[1])
```

Comment:
missing assignment?

Reply:
No, this is just testing whether the bounds can be transformed into floats. If it fails with a `ValueError`: could not convert string to float, it will fail the `try` and go to the `except` with 
```
raise ValueError(
    f"Censoring bound(s) for a {row[MEASUREMENT_TYPE]}"
    " measurement must be of type float."
) from e
```

Reply:
Hmmm you're right, one ValueError will trigger the other one. I'll make it explicit (put the try excepts exactly where the thing being tested is at).